src/App.tsx

```diff
@@ -10,33 +11,29 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]);
+
+  const onTabSelected = (tab: Tab) => {
+    const index = tabs.findIndex(t => t.id === tab.id);
+
+    if (index === -1) {
+      setSelectedTab(tabs[0]);
+    } else {
+      setSelectedTab(tabs[index]);
+    }
```

Comment:
would be better to calculate it inside the separate variable
```suggestion
  const newSelectedTab = index >= 0
    ? tabs[0]
    : tabs[index]
```